File: src/dataset/cpp/grade-school_exercism-johnngugi/gemini/GeminiFlash_grade_school_v2.cpp
#include "grade_school.h"
#include <algorithm>
#include <vector>
#include <map>
#include <string>

using namespace std;
using namespace grade_school;

map<int, vector<string>> school::roster() const
{
    return grades;
}

void school::add(const string& name, int grade)
{
    grades[grade].push_back(name);
    sort(grades[grade].begin(), grades[grade].end());
}

vector<string> school::grade(int grade) const
{
    auto it = grades.find(grade);
    if (it != grades.end()) {
        return it->second;
    } else {
        return {};
    }
}

File: src/dataset/typescript/grade-school_exercism-wackerow/gemini/GeminiFlash_grade-school_v2.ts
type Roster = Map<string, string[]>;

export default class School {
  private roster: Roster = new Map();

  public studentRoster(): Roster {
    const newRoster: Roster = new Map();
    for (const [grade, students] of this.roster) {
      newRoster.set(grade, students.slice()); // Efficiently create a copy
    }
    return newRoster;
  }

  public addStudent(name: string, grade: number): void {
    const gradeString = grade.toString();
    let gradeRoster = this.roster.get(gradeString);

    if (!gradeRoster) {
      gradeRoster = [];
      this.roster.set(gradeString, gradeRoster);
    }

    gradeRoster.push(name);
    gradeRoster.sort();
  }

  public studentsInGrade(grade: number): string[] {
    const gradeString = grade.toString();
    const gradeRoster = this.roster.get(gradeString);
    return gradeRoster ? gradeRoster.slice() : [];
  }
}